src/parquet/file/writer-internal.cc

```diff
@@ -142,7 +142,7 @@ ColumnWriter* RowGroupSerializer::NextColumn() {
   // Throws an error if more columns are being written
   auto col_meta = metadata_->NextColumnChunk();
 
-  if (current_column_writer_) { total_bytes_written_ = current_column_writer_->Close(); }
+  if (current_column_writer_) { total_bytes_written_ += current_column_writer_->Close(); }
```

Comment:
The `total_bytes_written_` is what is being used as the `total_byte_size` in the row group metadata. If you revert the `+=` to `=` the DCHECK (since the test suite now writes multi-column tables) fails